tests/listeners/very_slow_listener.py

```diff
@@ -0,0 +1,26 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import time
+
+from airflow.listeners import hookimpl
+
+
+@hookimpl
+def on_task_instance_success(previous_state, task_instance, session):
+    time.sleep(30)
```

Comment:
Have not run the unit tests but will this slow down unit testing by 30 seconds? Any way to mock the delay w/o a real sleep?

Reply:
@jscheffl not exactly - this is specifically too long to make the timeout mechanism reliably kill the process. However, since this is running real timeouts that measure time, _some_ seconds spend running those tests are reasonable. 

I've modified the values a bit and now `pytest tests/jobs/test_local_task_job.py` take
`66.58s` compared to `46.27s` - I don't think that's too bad.

Reply:
Maybe, we can do it with `time_machine` ? It's already used in a number of our tests and it's very easy to mock time using it

Reply:
Tried to naively use `frozen_sleep` and I don't think it plays nice with heartbeat mechanism. 

```
INFO     airflow.jobs.local_task_job_runner.LocalTaskJobRunner:local_task_job_runner.py:122 ::group::Pre task execution logs
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 420.04 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 750.01 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 665.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 560.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 330.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 220.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 345.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 360.00 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 460.01 seconds
INFO     airflow.jobs.job.Job:job.py:225 Heartbeat recovered after 2170.02 seconds
ERROR    airflow.jobs.local_task_job_runner.LocalTaskJobRunner:local_task_job_runner.py:215 Heartbeat time limit exceeded!
INFO     airflow.jobs.local_task_job_runner.LocalTaskJobRunner:local_task_job_runner.py:223 ::endgroup::
```

However, got it down to `51.44s`.